templates/client-daemonset.yaml

```diff
@@ -177,9 +205,16 @@ spec:
                 - -c
                 - consul leave
```

Comment:
I'm so glad you asked this! After some testing, I've found that consul clients already have `leave_on_terminate` set to true by default (as of consul 0.7) according to the [docs](https://www.consul.io/docs/agent/options.html#leave_on_terminate). I'll set that value to true for clients, so that it works even for an earlier version. That way we can delete the preStop hook.

For the servers though, the `preStop` hook doesn't work currently even without TLS. If I delete server pods, I'm seeing in the logs that it exits 1 instead of 0. I think it's because servers set `terminationGracePeriodSeconds` to 10s and so there's not enough time for it to properly exit. Unfortunately, leave_on_terminate has a hard-coded grace period of 15seconds which isn't enough for the servers. I'll keep the `preStop` hook there and increase `terminationGracePeriodSeconds` to 30sec (default).